src/sos/section_analyzer.py

```diff
@@ -409,7 +408,7 @@ def get_step_input(section, default_input):
         # expression ...
         step_input = sos_targets(_undetermined=stmt)
     finally:
-        [env.sos_dict.dict().pop(x) for x in svars]
+        old_values = [env.sos_dict.dict().pop(x) for x in svars]
```

Comment:
The same, the expression was used for its side effect (`.pop()`), so change to a for loop.